Feature flags methods fail with non-str distinct_id (regression?)

It looks like a breaking change.

`posthog.get_feature_flag` and `posthog.feature_enabled` suddenly stopped working with `distinct_id` that is an integer.
It used to work just fine.

`posthog.capture` still works with non-str `distinct_id`

Python 3.13
Posthog 3.25.0
Feature flag is enabled based on exact email:

<img width="768" alt="Image" src="https://github.com/user-attachments/assets/2babc536-de65-4e68-b1f4-33c3f99c3450" />

Stacktrace:

> [ERROR] [FEATURE FLAGS] Unable to get flag remotely: [PostHog] Failed to decode request: invalid JSON. Please check your request format and try again. (400)
> Traceback (most recent call last):
>   File ".../.venv/lib/python3.13/site-packages/requests/models.py", line 974, in json
>     return complexjson.loads(self.text, **kwargs)
>            ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.13/json/__init__.py", line 346, in loads
>     return _default_decoder.decode(s)
>            ~~~~~~~~~~~~~~~~~~~~~~~^^^
>   File "/usr/lib/python3.13/json/decoder.py", line 345, in decode
>     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
>                ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.13/json/decoder.py", line 363, in raw_decode
>     raise JSONDecodeError("Expecting value", s, err.value) from None
> json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File ".../.venv/lib/python3.13/site-packages/posthog/request.py", line 96, in _process_response
>     payload = res.json()
>   File ".../.venv/lib/python3.13/site-packages/requests/models.py", line 978, in json
>     raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
> requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File ".../.venv/lib/python3.13/site-packages/posthog/client.py", line 980, in get_feature_flag
>     flag_details, request_id = self._get_feature_flag_details_from_decide(
>                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
>         key, distinct_id, groups, person_properties, group_properties, disable_geoip
>         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>     )
>     ^
>   File ".../.venv/lib/python3.13/site-packages/posthog/client.py", line 1105, in _get_feature_flag_details_from_decide
>     resp_data = self.get_flags_decision(distinct_id, groups, person_properties, group_properties, disable_geoip)
>   File ".../.venv/lib/python3.13/site-packages/posthog/client.py", line 433, in get_flags_decision
>     resp_data = flags(
>         self.api_key, self.host, timeout=self.feature_flags_request_timeout_seconds, **request_data
>     )
>   File ".../.venv/lib/python3.13/site-packages/posthog/request.py", line 112, in flags
>     return _process_response(res, success_message="Feature flags evaluated successfully")
>   File ".../.venv/lib/python3.13/site-packages/posthog/request.py", line 100, in _process_response
>     raise APIError(res.status_code, res.text)
> posthog.request.APIError: [PostHog] Failed to decode request: invalid JSON. Please check your request format and try again. (400)

--- a/rust/feature-flags/src/flags/flag_request.rs
+++ b/rust/feature-flags/src/flags/flag_request.rs
@@ -1,11 +1,29 @@
 use std::collections::HashMap;
 
 use bytes::Bytes;
-use serde::{Deserialize, Serialize};
+use serde::{Deserialize, Deserializer, Serialize};
 use serde_json::Value;
 use tracing::instrument;
 
 use crate::api::errors::FlagError;
+
+fn deserialize_distinct_id<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum StringOrNumber {
+        String(String),
+        Number(i64),
+    }
+
+    let opt = Option::<StringOrNumber>::deserialize(deserializer)?;
+    Ok(opt.map(|val| match val {
+        StringOrNumber::String(s) => s,
+        StringOrNumber::Number(n) => n.to_string(),
+    }))
+}
 
 #[derive(Debug, Clone, Copy)]
 pub enum FlagRequestType {
@@ -21,7 +39,12 @@ pub struct FlagRequest {
         skip_serializing_if = "Option::is_none"
     )]
     pub token: Option<String>,
-    #[serde(alias = "$distinct_id", skip_serializing_if = "Option::is_none")]
+    #[serde(
+        alias = "$distinct_id",
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_distinct_id",
+        default
+    )]
     pub distinct_id: Option<String>,
     pub geoip_disable: Option<bool>,
     #[serde(default)]
@@ -157,6 +180,35 @@ mod tests {
             Ok(id) => assert_eq!(id, "alakazam"),
             _ => panic!("expected distinct id"),
         };
+    }
+
+    #[test]
+    fn numeric_distinct_id_is_returned_correctly() {
+        let json = json!({
+            "$distinct_id": 8675309,
+            "token": "my_token1",
+        });
+        let bytes = Bytes::from(json.to_string());
+
+        let flag_payload = FlagRequest::from_bytes(bytes).expect("failed to parse request");
+
+        match flag_payload.extract_distinct_id() {
+            Ok(id) => assert_eq!(id, "8675309"),
+            _ => panic!("expected distinct id"),
+        };
+    }
+
+    #[test]
+    fn missing_distinct_id_is_handled_correctly() {
+        let json = json!({
+            "token": "my_token1",
+        });
+        let bytes = Bytes::from(json.to_string());
+
+        let flag_payload = FlagRequest::from_bytes(bytes).expect("failed to parse request");
+
+        // First verify the field is None
+        assert_eq!(flag_payload.distinct_id, Option::<String>::None);
     }
 
     #[tokio::test]

--- a/rust/feature-flags/src/flags/flag_request.rs
+++ b/rust/feature-flags/src/flags/flag_request.rs
@@ -15,7 +15,7 @@ where
     #[serde(untagged)]
     enum StringOrNumber {
         String(String),
-        Number(i64),
+        Number(serde_json::Number),
     }
 
     let opt = Option::<StringOrNumber>::deserialize(deserializer)?;
@@ -209,6 +209,18 @@ mod tests {
 
         // First verify the field is None
         assert_eq!(flag_payload.distinct_id, Option::<String>::None);
+    }
+
+    #[test]
+    fn float_distinct_id_is_handled_correctly() {
+        let json = json!({
+            "$distinct_id": 123.45,
+            "token": "my_token1",
+        });
+        let bytes = Bytes::from(json.to_string());
+
+        let flag_payload = FlagRequest::from_bytes(bytes).expect("failed to parse request");
+        assert_eq!(flag_payload.distinct_id, Some("123.45".to_string()));
     }
 
     #[tokio::test]

